app/authProvider.js

```diff
@@ -84,4 +84,17 @@ export class AuthProvider {
       post_logout_redirect_uri: `${appBaseUri}/signout-callback-oidc`,
     });
   }
+
+  authorise() {
+    return (req, res, next) => {
+      if (!req.user) {
+        req.headers.referer = `${appBaseUri}${req.originalUrl}`;
```

Comment:
why do we need to do this here?

Reply:
We need this to get it to redirect back to the page we was trying to access after the login has completed. When I didn't have this line it would fail when it called `login` it would complain that `req.headers.referer` is undefined. So I explicitly set it here. Which seems to work but I'm not sure if this is the right way to do this.